### DISCUSSION TITLE
Async IO with State

### DISCUSSION THREAD

**Discussion by ghost** [louthy/language-ext#913] [General]
I'm trying to wrap my head around Reader, Writer, and State monads. I read [this comment](https://github.com/louthy/language-ext/issues/515#issuecomment-433991418) but I couldn't figure out how to do async IO. I'm not sure if it is possible or not.

So:
1) Is it possible to use async IO with State monad?
2) If yes, how?

Thanks in advance!

**Comment by louthy** [comment]
The `RWS`, `Reader`, `Writer`, and `State` monads are all synchronous.  Technically you could use `Task<State<S, A>>` and use the `BindT` extension to make it work, but it won't be too easy.  What's your use-case?  Because perhaps the `Aff<Env, A>` monad would work for you (asynchronous reader).

Of course you could roll your own, which wouldn't be too hard.  I can give some guidance on that als.
